PWGHF/DataModel/CandidateReconstructionTables.h

```diff
@@ -1531,6 +1531,178 @@ DECLARE_SOA_TABLE(HfCandScMcGen, "AOD", "HFCANDSCMCGEN", //!
+                           [](float xVtxP, float yVtxP, float xVtxS, float yVtxS, float err) -> float { return RecoDecay::distanceXY(std::array{xVtxP, yVtxP}, std::array{xVtxS, yVtxS}) / err; });
+DECLARE_SOA_COLUMN(D0ErrorDecayLength, d0errorDecayLength, float);
+DECLARE_SOA_COLUMN(D0ErrorDecayLengthXY, d0errorDecayLengthXY, float);
+DECLARE_SOA_DYNAMIC_COLUMN(D0CPA, d0cpa, //!
+                           [](float xVtxP, float yVtxP, float zVtxP, float xVtxS, float yVtxS, float zVtxS, float px, float py, float pz) -> float { return RecoDecay::cpa(std::array{xVtxP, yVtxP, zVtxP}, std::array{xVtxS, yVtxS, zVtxS}, std::array{px, py, pz}); });
+DECLARE_SOA_DYNAMIC_COLUMN(D0CPAXY, d0cpaXY, //!
+                           [](float xVtxP, float yVtxP, float xVtxS, float yVtxS, float px, float py) -> float { return RecoDecay::cpaXY(std::array{xVtxP, yVtxP}, std::array{xVtxS, yVtxS}, std::array{px, py}); });
+DECLARE_SOA_DYNAMIC_COLUMN(D0Ct, d0ct, //!
+                           [](float xVtxP, float yVtxP, float zVtxP, float xVtxS, float yVtxS, float zVtxS, float px, float py, float pz, double m) -> float { return RecoDecay::ct(std::array{px, py, pz}, RecoDecay::distance(std::array{xVtxP, yVtxP, zVtxP}, std::array{xVtxS, yVtxS, zVtxS}), m); });
+DECLARE_SOA_DYNAMIC_COLUMN(D0ImpactParameterXY, d0impactParameterXY, //!
+                           [](float xVtxP, float yVtxP, float zVtxP, float xVtxS, float yVtxS, float zVtxS, float px, float py, float pz) -> float { return RecoDecay::impParXY(std::array{xVtxP, yVtxP, zVtxP}, std::array{xVtxS, yVtxS, zVtxS}, std::array{px, py, pz}); });
+
+// Columns only for D* properties
+DECLARE_SOA_INDEX_COLUMN_FULL(ProngPi, prongPi, int, Tracks, "");                 // Softpi Index
+DECLARE_SOA_INDEX_COLUMN_FULL(ProngD0Cand, prondD0Cand, int, HfCand2Prong, "_0"); // D0 Index
+                                                                                  // SoftPiProng
+DECLARE_SOA_COLUMN(ImpParamSoftPiProng, impParamSoftPiProng, float);
+DECLARE_SOA_COLUMN(ErrorImpParamSoftPiProng, errorImpParamSoftPiProng, float);
+DECLARE_SOA_DYNAMIC_COLUMN(NormalisedImpParamSoftPiProng, normalisedImpParamSoftPiProng,
+                           [](float dca, float err) -> float { return dca / err; });
+DECLARE_SOA_COLUMN(PxSoftpiProng, pxSoftpiProng, float);
+DECLARE_SOA_COLUMN(PySoftpiProng, pySoftpiProng, float);
+DECLARE_SOA_COLUMN(PzSoftpiProng, pzSoftpiProng, float);
+// Dstar momenta
+DECLARE_SOA_EXPRESSION_COLUMN(DstarPx, dstarPx, float, 1.f * aod::hf_cand::pxProng0 + 1.f * aod::hf_cand::pxProng1 + 1.f * aod::hf_cand_dstar::pxSoftpiProng);
+DECLARE_SOA_EXPRESSION_COLUMN(DstarPy, dstarPy, float, 1.f * aod::hf_cand::pyProng0 + 1.f * aod::hf_cand::pyProng1 + 1.f * aod::hf_cand_dstar::pySoftpiProng);
+DECLARE_SOA_EXPRESSION_COLUMN(DstarPz, dstarPz, float, 1.f * aod::hf_cand::pzProng0 + 1.f * aod::hf_cand::pzProng1 + 1.f * aod::hf_cand_dstar::pzSoftpiProng);
```

Comment:
Fix the order `DstarPx` -> `PxDstar` and elsewhere.

Reply:
Is there any guideline for these conventions? It will help me in further coding.

Reply:
Yes, the guidelines are in the official analysis framework documentation.